infrastructures/openshift/src/test/java/org/eclipse/che/workspace/infrastructure/openshift/environment/OpenShiftEnvironmentParserTest.java

```diff
@@ -48,48 +50,53 @@
 import org.testng.annotations.Test;
 
 /**
- * Tests {@link OpenShiftEnvironmentParser}.
+ * Tests {@link OpenShiftEnvironmentFactory}.
  *
  * @author Anton Korneta
  */
 public class OpenShiftEnvironmentParserTest {
 
   private static final String YAML_RECIPE = "application/x-yaml";
 
-  private OpenShiftEnvironmentParser osEnvironmentParser;
+  private OpenShiftEnvironmentFactory osEnvironmentParser;
 
   @Mock private OpenShiftClientFactory factory;
   @Mock private OpenShiftClient client;
-  @Mock private InternalEnvironment internalEnvironment;
-  @Mock private InternalRecipe internalRecipe;
+  @Mock private Environment environment;
+  @Mock private Recipe recipe;
   @Mock private KubernetesListMixedOperation listMixedOperation;
   @Mock private KubernetesList validatedObjects;
+  @Mock private InstallerRegistry installerRegistry;
+  @Mock private RecipeRetriever recipeRetriever;
 
   @Captor private ArgumentCaptor<Warning> warningCaptor;
 
+  OpenShiftInternalEnvironment parsed;
+
   @Mock
   private RecreateFromServerGettable<KubernetesList, KubernetesList, DoneableKubernetesList>
       serverGettable;
 
   @BeforeMethod
   public void setup() {
     MockitoAnnotations.initMocks(this);
-    osEnvironmentParser = new OpenShiftEnvironmentParser(factory);
+    osEnvironmentParser =
+        new OpenShiftEnvironmentFactory(installerRegistry, recipeRetriever, factory);
     when(factory.create()).thenReturn(client);
     when(client.lists()).thenReturn(listMixedOperation);
     when(listMixedOperation.load(any(InputStream.class))).thenReturn(serverGettable);
     when(serverGettable.get()).thenReturn(validatedObjects);
-    when(internalEnvironment.getRecipe()).thenReturn(internalRecipe);
-    when(internalRecipe.getContentType()).thenReturn(YAML_RECIPE);
-    when(internalRecipe.getContent()).thenReturn("recipe content");
+    when(environment.getRecipe()).thenReturn(recipe);
+    when(recipe.getContentType()).thenReturn(YAML_RECIPE);
+    when(recipe.getContent()).thenReturn("recipe content");
   }
 
   @Test
   public void ignoreRoutesWhenRecipeContainsThem() throws Exception {
     final List<HasMetadata> objects = asList(new Route(), new Route());
     when(validatedObjects.getItems()).thenReturn(objects);
 
-    final OpenShiftEnvironment parsed = osEnvironmentParser.parse(internalEnvironment);
+    parsed = (OpenShiftInternalEnvironment) osEnvironmentParser.create(environment);
```

Comment:
I think `#create(Map<String, InternalMachineConfig>, InternalRecipe, List<Warning>)` should be used here. As `#create(InternalEnvironment` is a functional part of AbstractInternalEnvironmentParser and should be tested in the corresponding test and should be used while implementation testing